app/upgrades/v1-beta.4/bank_send.go

```diff
@@ -0,0 +1,237 @@
+package v1_beta4
+
+import (
+	"fmt"
+	"strconv"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	authvesting "github.com/cosmos/cosmos-sdk/x/auth/vesting/types"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+)
+
+func upgradeBankSend(
+	ctx sdk.Context,
+	authkeeper authkeeper.AccountKeeper,
+	bankkeeper bankkeeper.Keeper,
+	bank_send_list ResultList) error {
+	ctx.Logger().Info(fmt.Sprintf("upgrade :%s", UpgradeName))
+
+	total_allocate_coin := sdk.NewCoin(Denom, sdk.NewInt(0))
+	assumed_coin := sdk.NewCoin(Denom, sdk.NewInt(0))
+
+	// before get total supply
+	before_total_supply := bankkeeper.GetSupply(ctx, Denom)
+	ctx.Logger().Info(fmt.Sprintf("bank send : total supply[%d]", before_total_supply.Amount))
+
+	// others
```

Comment:
funding 'Others' must be the first operation in this upgrade since the recipients of the 'Others' need to have that fund to execute the distribution as we talked.